flytekit/extras/pydantic/transformer.py

```diff
@@ -0,0 +1,81 @@
+import json
+from typing import Type
+
+import msgpack
+from google.protobuf import json_format as _json_format
+from pydantic import BaseModel
+
+from flytekit import FlyteContext
+from flytekit.core.constants import MESSAGEPACK
+from flytekit.core.type_engine import TypeEngine, TypeTransformer, TypeTransformerFailedError
+from flytekit.loggers import logger
+from flytekit.models import types
+from flytekit.models.literals import Binary, Literal, Scalar
+from flytekit.models.types import LiteralType, TypeStructure
+
+
+class PydanticTransformer(TypeTransformer[BaseModel]):
+    def __init__(self):
+        super().__init__("Pydantic Transformer", BaseModel, enable_type_assertions=False)
+
+    def get_literal_type(self, t: Type[BaseModel]) -> LiteralType:
+        schema = t.model_json_schema()
```

Comment:
In a future PR, can we add some unit tests to ensure that we're correctly extracting [default values](https://github.com/flyteorg/flyte/issues/5887#issuecomment-2436565600) into the schema?

Reply:
Yes we can

Reply:
I'll ask more question about this unit tests PR if not totally understand, thank you